ssl/quic/quic_port.c

```diff
@@ -848,8 +848,10 @@ static int marshal_validation_token(QUIC_VALIDATION_TOKEN *token,
     BUF_MEM *buf_mem = BUF_MEM_new();
 
     if (buffer == NULL || buf_mem == NULL
-        || (token->is_retry != 0 && token->is_retry != 1))
+        || (token->is_retry != 0 && token->is_retry != 1)) {
+        BUF_MEM_free(buf_mem);
         return 0;
+    }
```

Comment:
Looks like I can see a side issue here.. WPACKET_cleanup() is not called on the success path?

Reply:
`WPACKET_finish` frees memory resources for the WPACKET: https://github.com/openssl/openssl/blob/master/include/internal/packet.h#L756